### ISSUE TITLE
mypy error "Unexpected keyword argument" on pydantic dataclass InitVar field

### ISSUE THREAD

**Issue by brianmedigate** [pydantic/pydantic#5161]
### Initial Checks

- [X] I have searched GitHub for a duplicate issue and I'm sure this is something new
- [X] I have searched Google & StackOverflow for a solution and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this is a bug
- [X] I am confident that the issue is with pydantic (not my code, or another library in the ecosystem like [FastAPI](https://fastapi.tiangolo.com) or [mypy](https://mypy.readthedocs.io/en/stable))

### Description

Since upgrading to pydantic 1.10.6, mypy started complaining about a kwarg being passed to the constructor of a dataclass if the field is an `InitVar`.

I'm guessing it might be related to https://github.com/pydantic/pydantic/pull/5111

```
$ mypy test.py
test.py:10: error: Unexpected keyword argument "foo" for "MyDataclass"  [call-arg]
Found 1 error in 1 file (checked 1 source file)
```

### Example Code

```Python
from pydantic.dataclasses import dataclass
from dataclasses import InitVar

@dataclass
class MyDataclass:
    foo: InitVar[str]
    bar: str

MyDataclass(foo="foo", bar="bar")
```

### Python, Pydantic & OS Version

```Text
pydantic version: 1.10.6
            pydantic compiled: True
                 install path: /home/brian/repos/medigator/venv/lib/python3.8/site-packages/pydantic
               python version: 3.8.10 (default, Nov 14 2022, 12:59:47)  [GCC 9.4.0]
                     platform: Linux-5.15.0-67-generic-x86_64-with-glibc2.29
     optional deps. installed: ['dotenv', 'email-validator', 'typing-extensions']
```

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/usage/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/usage/exporting_models/) - `.dict()` and `.json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/usage/schema/)
- [X] [Dataclasses](https://docs.pydantic.dev/usage/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/usage/model_config/)
- [ ] [Field Types](https://docs.pydantic.dev/usage/types/) - adding or changing a particular data type
- [ ] [Function validation decorator](https://docs.pydantic.dev/usage/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/usage/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/usage/models/) - `construct()`, pickling, private attributes, ORM mode
- [X] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by samuelcolvin** [comment]
This won't be related to #2309, rather it will be related to #5111.

@cdce8p @dmontagu any ideas?

**Comment by dmontagu** [comment]
@brianmedigate what is your mypy config? In particular, are you using the mypy plugin?

**Comment by brianmedigate** [comment]
great, thank you!

**Comment by cdce8p** [comment]
Possible fix: #5162
Until now, the plugin manually applied the dataclass transform to pydantic dataclasses. This isn't necessary anymore with mypy `1.1.1` and the `@dataclass_transform` decorator as it also applies the same transform.

In fact, doing it twice results in the `InitVar` issue seen here.

**Comment by brianmedigate** [comment]
thank you @cdce8p @dmontagu and @samuelcolvin, you're awesome! :1st_place_medal:
